apps/internal/exported/exported.go

```diff
@@ -0,0 +1,14 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+// package exported contains internal types that are re-exported from a public package
+package exported
+
+// AssertionRequestOptions has information required to generate a client assertion
+type AssertionRequestOptions struct {
```

Comment:
any reason not to just define this as part of the accesstokens package and then do the same re-export from there?

That is closer to the existing patterns followed in this project I believe.

Reply:
We adopted this convention in the Azure SDK to make pubternal API more explicit. This helps SDK developers avoid breaking public APIs with apparently internal changes, and improves documentation by preventing pkgsite from showing users stuff that may not be exported ([like this](https://pkg.go.dev/github.com/AzureAD/microsoft-authentication-library-for-go/apps/confidential#AuthResult)).

It is a new pattern for MSAL though. I don't mind moving the current aliased types into `exported` in another PR or, if you feel strongly about maintaining the current pattern, following it here.

Reply:
makes sense, thanks for explaining.